rust_src/src/alloc.rs

```diff
@@ -95,4 +96,19 @@ pub fn record(args: &mut [LispObject]) -> LispObject {
     }
 }
 
+/// Make a copy of object OBJ in pure storage.
+/// Recursively copies contents of vectors and cons cells.
+/// Does not copy symbols.  Copies strings without text properties.
+#[lisp_fn(name = "purecopy", c_name = "purecopy")]
+pub fn purecopy_lisp(obj: LispObject) -> LispObject {
```

Comment:
Clippy complains that both of the arms of the if/else result in `obj`.

Reply:
I thought about changing that, but the separate arms don't relate to each other, they just happen to return the same value.

Reply:
That is a clippy error, not a warning.

Reply:
You could do this:
```
if unsafe { globals.Vpurify_flag.is_nil() } {
    obj
}

if obj.is_marker() || obj.is_overlay() || obj.is_symbol() {
    // Can't purify those.
    obj
} else {
    unsafe { c_purecopy(obj) }
}
```